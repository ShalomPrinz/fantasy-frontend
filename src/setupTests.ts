// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { ReactElement } from 'react';

import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

export type TestRendererJSON = null | ReactTestRendererJSON | ReactTestRendererJSON[];
export const getJSON = (Component: ReactElement) => renderer.create(Component).toJSON()
