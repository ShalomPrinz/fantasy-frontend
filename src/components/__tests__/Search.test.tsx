import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Search } from '../';
import { getJSON, TestRendererJSON } from '../../setupTests'

let tree: TestRendererJSON = null;

beforeEach(() => {
    tree = null
})

describe('Search', () => {
    it('should render Search component', () => {
        const value = "";
        const onChange = () => {};
    
        tree = getJSON(<Search onChange={onChange} value={value} />);
    
        expect(tree).toMatchSnapshot();
    })

    describe('when input changes', () => {
        it('should call onChange with the new input as argument', () => {
            let value = ""
            const onChange = (v: string) => value += v
            render(<Search onChange={onChange} value={value} />)
            
            const Component = screen.getByRole('textbox')
            const input = "NEW INPUT"
            userEvent.type(Component, input)

            expect(value).toBe(input)
        })
    })
})
