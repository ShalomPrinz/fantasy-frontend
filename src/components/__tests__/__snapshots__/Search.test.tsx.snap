// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render Search component 1`] = `
<input
  className="form-control my-3 fs-1 p-4"
  onChange={[Function]}
  placeholder="Search..."
  type="text"
  value=""
/>
`;
